metaflow/plugins/events_decorator.py

```diff
@@ -0,0 +1,298 @@
+import json
+import time
+import re
+
+from metaflow import current
+from metaflow.decorators import FlowDecorator
+from metaflow.exception import MetaflowException
+from metaflow.util import is_stringish
+
+# TODO: Support dynamic parameter mapping through a context object that exposes
+#       flow name and user name similar to parameter context
+
+
+class TriggerDecorator(FlowDecorator):
+    """
+    Specifies the event(s) that this flow depends on.
+
+    ```
+    @trigger(event='foo')
+    ```
+    or
+    ```
+    @trigger(events=['foo', 'bar'])
+    ```
+
+    Additionally, you can specify the parameter mappings
+    to map event payload to Metaflow parameters for the flow.
+    ```
+    @trigger(event={'name':'foo', 'parameters':{'my_param': 'event_field'})
+    ```
+    or
+    ```
+    @trigger(events=[{'name':'foo', 'parameters':{'my_param_1': 'event_field_1'},
+                     {'name':'bar', 'parameters':{'my_param_2': 'event_field_2'}])
```

Comment:
As a tiny tiny nit: black imposes " instead of ' so we could also have that in our doc.